alodataset/kitti_tracking.py

```diff
@@ -213,8 +195,9 @@ def getitem(self, idx: int) -> Dict[str, Frame]:
             bounding_box = BoundingBoxes2D(boxes2d, boxes_format="xyxy", absolute=True, frame_size=left_frame.HW)
             bounding_box.append_labels(labels, "track_id")
             bounding_box.append_labels(Labels(categories, labels_names=LABELS), "categories")
-            boxe3d = BoundingBoxes3D(boxes3d)
-            # TODO : boxes3D should also have labels
+            boxe3d = BoundingBoxes3D(
+                boxes3d, labels={"track_id": labels, "categories": Labels(categories, labels_names=LABELS)}
```

Comment:
A same instance of `Labels` should not be the child of two different AugmentedTensor instance. Here `labels` is added as a child to both `bounding_box` and `boxe3d`. You should create a copy of it with `labels.clone()` instead of using the same instance.

Also, you have a condition for the bounding boxes 3D (you add the box only if the category is not "DontCare"). It is possible for some objects you only add a box 2D but not 3D. Therefore, the `Labels` objects for boxes2D and boxes3D are not exactly the same.

Is there a reason why you remove "DontCare" only for 3D boxes ?

Reply:
The objects with label `DontCare` doesn't have a correct box3d.